python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -154,6 +159,120 @@ def repository_name(self) -> Optional[str]:
         return self._repository_name
 
 
+class MultiAssetSensorEvaluationContext(SensorEvaluationContext):
+    """The context object available as the argument to the evaluation function of a :py:class:`dagster.MultiAssetSensorDefinition`.
```

Comment:
An issue I'm having with this new context is that the `SensorEvaluationContext` expects the cursor to be a string, but in this multi asset case, the cursor is a dict. it's screwing up a bunch of mypy checks and our internal `check` calls. I \~think\~ it makes sense to keep `MultiAssetSensorEvaluationContext` as a subclass of `SensorEvaluationContext` so that changes to the shared functions only need to be made in one place. One option would be to make the type for cursor in `SensorEvaluationContext` a `Union[str, Mapping[AssetKey, str]]` but i'm not sure if that would screw up backcompat or something. thoughts?

Reply:
Given that the cursors function differently, I think making it so that `MultiAssetSensorEvaluationContext` doesn't inherit from `SensorEvaluationContext` might be the most sensible path.